docs/src/main/asciidoc/security.adoc

```diff
@@ -456,5 +457,15 @@ void someTestMethod() {
 ----
 
 This will run the test with an identity with the given username and roles. Note that these can be combined, so you can
-disable authorisation and also provide an identity to run the test under, which can be userful if the endpoint expects an
-identity to be present.
\ No newline at end of file
+disable authorization while also providing an identity to run the test under, which can be useful if the endpoint expects an
+identity to be present.
+
+[WARNING]
+====
+The feature is only available for `@QuarkusTest` and will **not** work on a `@NativeImageTest`.
+====
+
+=== Mixing security tests
+
+If it becomes necessary to test security features using both `@TestSecurity` and a real authentication mechanism (like Basic Auth), then that authentication mechanism
```

Comment:
So this would only be an issue for basic auth, as it is the fallback if no other authentication mechanisms are defined. Because it uses @DefaultBean installing the test authentication provider will disable the basic auth mechanism. Can you change this slightly to explain that it is only an issue with 'implicit' basic auth (i.e. when no auth mechanisms have been explicitly configured).

Reply:
Sure yeah